### PR TITLE
coolprop-if97: submission of IF97 port

### PR DESCRIPTION
**Pull request by jsalort**
#### Description

Submission of the IF97 port.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.1 21G217 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] checked your Portfile with `port lint --nitpick`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: science/coolprop-if97/Portfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake  1.1
+PortGroup           compiler_blacklist_versions 1.0
+PortGroup           conflicts_build 1.0
```
Comment (reneeotten) [inline science/coolprop-if97/Portfile:7]: you don't use the last two PortGroups

File: science/coolprop-if97/Portfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                    sha256  5d24a8ca0aeefb3a5085c9c10c92cf7b8e8c43a56953f5b5f7ced49cc03f6d1f \
+                    size    3893915
+
+maintainers         {juliensalort.org:mp @jsalort} openmaintainer
+categories-append   science
+
+description         Implementation of the IF97 formulation for the properties of pure water substance.
+long_description    Implementation of the IF97 formulation for the properties of pure water substance by \
+                    Ian Bell and the CoolProp team.
+homepage            http://www.coolprop.org
```
Comment (reneeotten) [inline science/coolprop-if97/Portfile:25]: can this use "https"?

File: science/coolprop-if97/Portfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+cmake.build_type    Release
```
Comment (reneeotten) [inline science/coolprop-if97/Portfile:27]: does the default "MacPorts" as set by the cmake PG fail?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: science/coolprop-if97/Portfile (hunk 1 above)
Comment (jsalort) [reply science/coolprop-if97/Portfile]: Right.

File: science/coolprop-if97/Portfile (hunk 2 above)
Comment (jsalort) [reply science/coolprop-if97/Portfile:23]: Apparently not.
```bash
curl -v https://www.coolprop.org
*   Trying 204.68.111.100:443...
* connect to 204.68.111.100 port 443 failed: Connection refused
* Failed to connect to www.coolprop.org port 443 after 189 ms: Couldn't connect to server
* Closing connection 0
```

File: science/coolprop-if97/Portfile (hunk 3 above)
Comment (jsalort) [reply science/coolprop-if97/Portfile]: Actually it doesn't. I have removed this line. Thanks.

**Pull request merged**